kotlin/codegen/pom.xml

```diff
@@ -176,6 +186,11 @@
                   <pattern>com.squareup.kotlinpoet</pattern>
                   <shadedPattern>com.squareup.moshi.kotlinpoet</shadedPattern>
                 </relocation>
+                <!-- Repackage Kotlinx-metadata until its API is stable. -->
+                <relocation>
+                  <pattern>kotlinx.metadata.jvm</pattern>
```

Comment:
I suppose you need to relocate `kotlinx.metadata` here, not `kotlinx.metadata.jvm`. I think this may fix the ServiceLoader issue.

Reply:
Unfortunately this happens even if I disable shading